pensieve/analysis.py

```diff
@@ -0,0 +1,100 @@
+import mozanalysis.metrics.desktop as mmd
+from mozanalysis.experiment import Experiment, TimeLimits
+from mozanalysis.bq import BigQueryContext
+from datetime import datetime, timedelta
+import pytz
+import re
+
+
+class Analysis:
+    """
+    Wrapper for analysing experiments.
+    """
+
+    # todo: where should this come from? experiments, metrics?
+    ANALYSIS_PERIOD = 7  # 1 week
+
+    # list of standard metrics to be computed
+    STANDARD_METRICS = [
+        mmd.active_hours,
+        mmd.uri_count,
+        mmd.ad_clicks,
+        mmd.search_count,
+    ]
+
+    def __init__(self, project, dataset):
+        self.project = project
+        self.dataset = dataset
+        self.bq_context = BigQueryContext(project_id=project, dataset_id=dataset)
+
+    def _current_date(self):
+        """Returns the current date with UTC timezone and time set to 00:00:00."""
+        return datetime.combine(datetime.today(), datetime.min.time()).replace(tzinfo=pytz.utc)
+
+    def _should_analyse_experiment(self, experiment, current_date):
```

Comment:
The logic here is skipping the enrollment period, which we often take to be 8 "dates" of data -- the first seven-day analysis window only ends after the last client to enroll has had seven days of exposure.

Reply:
We should see if we can make mozanalysis do this check; it already knows this logic.

Reply:
Interesting. Is this related to the `proposed_enrollment` attribute that is available for experiments? I don't seem to be able to find any logic related to this in mozanlaysis

Reply:
It's not -- probably this involves interacting with the TimeLimits class somehow. One way might be "see whether TimeLimits produces a different number of analysis windows if you run it on yesterday's date vs today's date" which feels a little silly but probably works. The date math is just a little subtle and I'm wary of replicating it. I can be responsible for this but probably not until Wednesday; we can land this PR first if you'd like to make this my problem and want to push ahead :)

Reply:
Yeah, I think it'd be nice if you could fix that in a follow-up. For now it might be nice to just have something running.